packages/kit/src/core/dev/plugin.js

```diff
@@ -123,14 +127,27 @@ export async function create_plugin(config, cwd) {
 
 							return {
 								type: 'endpoint',
-								pattern: route.pattern,
-								params: get_params(route.params),
+								pattern,
+								names,
+								types,
 								load: async () => {
 									const url = path.resolve(cwd, route.file);
 									return await vite.ssrLoadModule(url);
 								}
 							};
-						})
+						}),
+						validators: async () => {
+							/** @type {Record<string, import('types').ParamValidator>} */
+							const validators = {};
+
+							for (const key in manifest_data.validators) {
+								const url = path.resolve(cwd, manifest_data.validators[key]);
+								const module = await vite.ssrLoadModule(url);
+								validators[key] = module.validate;
```

Comment:
throw an error if validate doesn't exist on the loaded module?

Reply:
good catch — fixed